src/utils/Assets.ts

```diff
@@ -0,0 +1,191 @@
+): Promise<string> {
+  if (!VALID_TEMPLATE_INDEXES.includes(templateIndex)) {
+    throw new Error('Invalid template index: ' + templateIndex)
+  }
+
+  const chainID = (await owner.provider.getNetwork()).chainId
+
+  const config = new ConfigHelper().getConfig(parseInt(String(chainID)))
+
+  const nft = new Nft(owner, chainID)
+
+  const nftFactory = new NftFactory(nftContractAddress, owner)
+
+  // get nft owner
+  const account = await owner.getAddress()
+
+  // from hex to number format
+  ddo.chainId = parseInt(chainID.toString(10))
+  const nftParamsAsset: NftCreateData = {
+    name,
+    symbol,
+    templateIndex,
+    tokenURI: 'aaa',
+    transferable: true,
+    owner: account
+  }
+  const datatokenParams: DatatokenCreateParams = {
+    templateIndex,
+    cap: '100000',
+    feeAmount: '0',
+    paymentCollector: account,
+    feeToken: providerFeeToken || config.oceanTokenAddress,
+    minter: account,
+    mpFeeAddress: ZERO_ADDRESS
+  }
+
+  // include fileObject in the DT constructor
+  if (templateIndex === 4) {
+    datatokenParams.filesObject = filesObject
+  }
+
+  let bundleNFT
+
+  if (!ddo.stats?.price?.value) {
+    bundleNFT = await nftFactory.createNftWithDatatoken(nftParamsAsset, datatokenParams)
+  } else if (ddo.stats?.price?.value === '0') {
+    const dispenserParams: DispenserCreationParams = {
+      dispenserAddress: config.dispenserAddress,
+      maxTokens: '1',
+      maxBalance: '100000000',
+      withMint: true,
+      allowedSwapper: ZERO_ADDRESS
+    }
+    bundleNFT = await nftFactory.createNftWithDatatokenWithDispenser(
+      nftParamsAsset,
+      datatokenParams,
+      dispenserParams
+    )
+  } else {
+    // fixed price
+    const fixedPriceParams: FreCreationParams = {
+      fixedRateAddress: config.fixedRateExchangeAddress,
+      baseTokenAddress: config.oceanTokenAddress,
+      owner: account,
+      marketFeeCollector: account,
+      baseTokenDecimals: 18,
+      datatokenDecimals: 18,
+      fixedRate: ddo.stats.price.value,
+      marketFee: '0',
+      allowedConsumer: account,
+      withMint: true
+    }
+    bundleNFT = await nftFactory.createNftWithDatatokenWithFixedRate(
+      nftParamsAsset,
+      datatokenParams,
+      fixedPriceParams
+    )
+  }
+
+  const trxReceipt = await bundleNFT.wait()
+  // events have been emitted
+  const nftCreatedEvent = getEventFromTx(trxReceipt, 'NFTCreated')
+  const tokenCreatedEvent = getEventFromTx(trxReceipt, 'TokenCreated')
+
+  const nftAddress = nftCreatedEvent.args.newTokenAddress
+  const datatokenAddressAsset = tokenCreatedEvent.args.newTokenAddress
+  // create the files encrypted string
+  assetUrl.datatokenAddress = datatokenAddressAsset
+  assetUrl.nftAddress = nftAddress
+  // TODO if template 4 no need to encrypt it??
+  if (templateIndex !== 4) {
+    ddo.services[0].files = await ProviderInstance.encrypt(assetUrl, chainID, providerUrl)
+  } else {
+    ddo.services[0].files = assetUrl
```

Comment:
if filesObject is stored in the datatoken contract, there is no point of keeping it in ddo.  Especially unencrypted